app/models/bundle_instance.rb

```diff
@@ -0,0 +1,12 @@
+class BundleInstance < ApplicationRecord
+  belongs_to :bundle
+  belongs_to :site
+  has_many :application_instances
+  has_many :applications, through: :bundle
+
+  def create_application_instances
+    bundle&.applications&.map do |app|
```

Comment:
Is there a reason for doing a `.map` here? Do you need an array returned of all these? If not, you should consider doing a `.each` instead.